### PR TITLE
API Backport security extensions backend to SilverStripe 3

### LINKED ISSUE #10: Backport to SS3 (backend and frontend)
Required to be done before silverstripe/mfa can be backported for SilverStripe 3

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #10

### CODE REVIEW

File: src/Extension/SudoModeOnLoginExtension.php [hunk 1]
```diff
@@ -2,26 +2,23 @@
...
+        $session = Controller::curr()->getSession();
+        if ($session) {
             /** @var SudoModeServiceInterface $service */
             $service = Injector::inst()->get(SudoModeServiceInterface::class);
-
-            /** @var HTTPRequest $request */
-            $request = Injector::inst()->get(HTTPRequest::class);
-
-            $service->activate($request->getSession());
+            $service->activate(Controller::curr()->getSession());
```
Comment (robbieaverill) [inline src/Extension/SudoModeOnLoginExtension.php]:
```suggestion
            $service->activate($session);
```

File: tests/php/Control/SudoModeControllerTest.php [hunk 2]
```diff
@@ -72,7 +73,7 @@ public function testActivateSudoModeWithValidCredentials()
 
         $this->assertSame(200, $activateResponse->getStatusCode());
         $result = json_decode((string) $activateResponse->getBody(), true);
-        $this->assertTrue($result['result'], 'Should have activated sudo mode');
+        $this->assertTrue($result['result'], 'Should have activated sudo mode' . print_r($result, true));
```
Comment (robbieaverill) [inline tests/php/Control/SudoModeControllerTest.php]:
```suggestion
        $this->assertTrue($result['result'], 'Should have activated sudo mode');
```

File: tests/php/Extension/MemberExtensionTest.php [hunk 3]
```diff
@@ -39,15 +40,15 @@ public function testUserCannotRequireOthersToPasswordChangeOnNextLogIn()
     {
         /** @var Member&MemberExtension $targetMember */
         $targetMember = $this->objFromFixture(Member::class, 'anyone');
-        $this->logInAs('someone');
+        $targetMember->logIn();
```
Comment (robbieaverill) [inline tests/php/Extension/MemberExtensionTest.php]: Should be `someone`, not `anyone`

**Review by ScopeyNZ — APPROVED** [review]
It works. I'm getting a weird bug that the state of the checkbox doesn't reflect that a password change is expected - but I'm not sure if that was existing.

Either way I'm just going to merge now.

**Pull request merged**